File: Python/121_best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max = 0
        minPrice = prices[0]

        # Find highest peek for every lowest point
        for i in prices:
            if i < minPrice:
                minPrice = i
            elif i - minPrice > max:
                max = i - minPrice

        return max

File: C++/121_best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max = 0;
        int minPrice = prices[0];
            
        // Find highest peek for every lowest point
        for(int i = 0; i < prices.size(); i++){
            if(prices[i] < minPrice){
                minPrice = prices[i];
            }
            else if(prices[i] - minPrice > max){
                max = prices[i] - minPrice;
            }
        }
        
        return max;
    }
};